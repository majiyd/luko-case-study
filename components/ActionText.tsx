import React from 'react'
import { StyleSheet, Text, Pressable, TextStyle } from 'react-native'
import Colors from '../constants/Colors';

type Props = {
  children: React.ReactNode;
  style?: TextStyle;
  onPress: () => void;
  isDisabled?: boolean;
};
const ActionText = ({children, style, onPress, isDisabled=false}: Props) => {
  return (
    <Pressable hitSlop={15} onPress={!isDisabled ? onPress : () => {}}>
      <Text style={[styles.text, style, {color: isDisabled ? Colors.light.gray300: Colors.light.bluko500}]}>{children}</Text>
    </Pressable>
  )
}

const styles = StyleSheet.create({
  text: {
    fontSize: 17,
    lineHeight: 24
  }
})
export default ActionText
