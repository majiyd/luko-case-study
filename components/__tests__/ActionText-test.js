import * as React from "react";
import renderer from "react-test-renderer";
import ActionText from "../ActionText";

const onPress = jest.fn();

describe("ActionText", () => {
  it("ActionText renders correctly", () => {
    const tree = renderer.create(<ActionText>Add</ActionText>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  // it("Disable ActionText is not pressable", () => {

  // })
});
