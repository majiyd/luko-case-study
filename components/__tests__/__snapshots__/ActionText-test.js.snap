// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActionText ActionText renders correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  hitSlop={
    Object {
      "bottom": 15,
      "left": 15,
      "right": 15,
      "top": 15,
    }
  }
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
>
  <Text
    style={
      Array [
        Object {
          "fontSize": 17,
          "lineHeight": 24,
        },
        undefined,
        Object {
          "color": "#2D50E6",
        },
      ]
    }
  >
    Add
  </Text>
</View>
`;
